src/ucp/core/ucp_ep.h

```diff
@@ -306,10 +306,19 @@ typedef struct ucp_ep {
  * Status of protocol-level remote completions
  */
 typedef struct {
-    ucs_queue_head_t              reqs;         /* Queue of flush requests which
-                                                   are waiting for remote completion */
-    uint32_t                      send_sn;      /* Sequence number of sent operations */
-    uint32_t                      cmpl_sn;      /* Sequence number of completions */
+    union {
+        struct {
+            ucs_queue_head_t              reqs;         /* Queue of flush requests which
+                                                           are waiting for remote completion */
+            uint32_t                      send_sn;      /* Sequence number of sent operations */
+            uint32_t                      cmpl_sn;      /* Sequence number of completions */
+        } sw;
+
+        struct {
+            ucp_request_t                 *req;         /* Flush request which is
+                                                           used in close protocol */
+        } close;
```

Comment:
IMHO close.req should be added to the union on ep  https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_ep.h#L329
this way, it would also have own state flag for assertion purpose (like UCP_EP_FLAG_FLUSH_STATE_VALID)